### PR TITLE
Defined desired behavior better in tests

### PR DESCRIPTION
**Pull request by samdark**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ❌
| Breaks BC?    | ✔️
| Tests pass?   | ❌
| Fixed issues  | -

### CODE REVIEW

**Comment by samdark** [comment]
@yiiliveext

File: tests/Unit/AbstractFactoryTest.php [hunk 1]
```diff
@@ -33,7 +33,10 @@ public function testCreateByAlias(): void
         $this->assertInstanceOf(EngineMarkOne::class, $two);
     }
 
-    public function testSingleton()
+    /**
+     * TODO: this is weird. Do we need such behavior?
```
Comment (hiqsol) [inline tests/Unit/AbstractFactoryTest.php]: Why not? Alternatives?

Comment (viktorprogger) [reply tests/Unit/AbstractFactoryTest.php]: I expect every `create` call to return an instance in the state I've defined. So I suggest to clone the defined instance.

Comment (samdark) [reply tests/Unit/AbstractFactoryTest.php]: Yes. A factory that returns the same instance is weird.

**Pull request merged**